cmd/nats-blackbox-exporter/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/snapp-incubator/nats-blackbox-exporter/internal/config"
+	"github.com/snapp-incubator/nats-blackbox-exporter/internal/logger"
+	"go.uber.org/zap"
+)
+
+var publishInterval = 2 * time.Second
+
+func main() {
+	cfg := config.New()
+
+	logger := logger.New(cfg.Logger)
+
+	nc, err := nats.Connect(cfg.NATS.URL)
+	if err != nil {
+		logger.Fatal("nats connection failed", zap.Error(err))
+	}
+
+	logger.Info("nats connection successful",
+		zap.String("connected-addr", nc.ConnectedAddr()),
+		zap.Strings("discovered-servers", nc.DiscoveredServers()))
+
+	nc.SetDisconnectErrHandler(func(_ *nats.Conn, err error) {
```

Comment:
Disruption in connections ( including disconnecting and reconnecting ) could be sign of unhealthy cluster. We can set metrics for these values too ( in future when you wanna expose metrics )

Reply:
Yes, I totally agree we can add prometheus for metrics.